packages/repository/src/repository-mixin.ts

```diff
@@ -71,6 +72,32 @@ export function RepositoryMixin<T extends Class<any>>(superClass: T) {
       this.bind(repoKey).toClass(repo);
     }
 
+    /**
+     * Add the dataSource to this application.
+     *
+     * @param dataSource The dataSource to add.
+     * @param name The binding name of the datasource; default to dataSource.name
+     *
+     * ```ts
+     *
+     * const ds: juggler.DataSource = new DataSourceConstructor({
+     *   name: 'db',
+     *   connector: 'memory',
+     * });
+     *
+     * app.dataSource(ds);
+     *
+     * The datasource can be injected with
+     * constructor(@inject('datasource.db') protected datasource: DataSourceType) {
+     *
+     * }
+     * ```
+     */
+    dataSource(dataSource: juggler.DataSource, name?: string) {
+      const dataSourceKey = `datasource.${name || dataSource.name}`;
```

Comment:
IIRC, we are using plural form for prefixes, e.g. `controllers.*`. I think we should be using `datasources.*` prefix for consistency.

@raymondfeng thoughts?

Reply:
Let us keep the style with the prefix `datasources.*`